server.go

```diff
@@ -411,30 +411,23 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq, inbound
 
 	// Now that we've established a connection, create a peer, and
 	// it to the set of currently active peers.
-	peer, err := newPeer(conn, s, peerAddr, false)
+	p, err := newPeer(conn, connReq, s, peerAddr, inbound)
 	if err != nil {
 		srvrLog.Errorf("unable to create peer %v", err)
-		s.connMgr.Remove(peer.connReq.ID())
-		peer.Disconnect()
-		conn.Close()
+		p.Disconnect()
```

Comment:
There are 2 reasons for such change:
- `Disconnect` function _already_ has this functionality.
-  Two bugs were introduced before, because we are trying to write the same code:
  - I forgot to remove connection, which you fix.
  - We forgot to add check for nil, before removing connection.
These bugs might be successfully avoided if we just use the `Disconnect` function.